wiring/src/spark_wiring_cellular_printable.cpp

```diff
@@ -80,18 +77,49 @@ float CellularSignal::getQualityValue() const
     return 0.0f;
 }
 
-// TODO: remove once rssi/qual are removed
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 size_t CellularSignal::printTo(Print& p) const
 {
     size_t n = 0;
-    n += p.print((*this).rssi, DEC);
+    n += p.print(this->getStrengthValue(), 2);
     n += p.print(',');
-    n += p.print((*this).qual, DEC);
+    n += p.print(this->getQualityValue(), 2);
     return n;
 }
-#pragma GCC diagnostic pop
+
+bool CellularSignal::isAccessTechnologyValid() const
+{
+    return (sig_.rat != NET_ACCESS_TECHNOLOGY_NONE);
+}
+
+bool CellularSignal::isSignalStrengthValid() const
+{
+    return (sig_.rssi != std::numeric_limits<int32_t>::min());
```

Comment:
`Cellular.RSSI()` [returns](https://github.com/particle-iot/device-os/blob/ch62644/remove_celluluar_rssi/wiring/src/spark_wiring_cellular.cpp#L31) a default-initialized `CellularSignal` object on certain errors, which has `rssi` and `qual` set to 0. `isSignalStrengthValid()` and `isSignalQualityValid()` called on such an object will incorrectly return `true`.

Reply:
It seems like we have an issue here with the API where strength/quality values that error are set to 0 instead of -1.  Of course -1 is a valid value for strength/quality so it makes sense.  This is why the previous RSSI().rssi error values were positive values 1 and 2.  We still use this `sig_.qual != std::numeric_limits<int32_t>::min())` in isValid().  Perhaps we need to introduce a new error value for these APIs so that they can be differentiated between default-initialized, valid, and error?

Reply:
We touched this topic in one of the comment threads here. Let's try not to introduce any special error values for user facing APIs where possible.

There also shouldn't be any issue as of right either, as one of the requirements for `CellularSignal` or a to be valid is a valid RAT. If we do introduce distinct methods to check if a particular part is valid, we can then make sure that underlying `cellular_signal_t` is initialized by default to invalid values.

Reply:
Sounds good, I agree.  I think if we did bring back the isX() methods, we could protect them with `sig_.rat != NET_ACCESS_TECHNOLOGY_NONE &&` like all other APIs use.